docs/DeveloperGuide.md

```diff
@@ -1,38 +1,433 @@
+
+### 2.1. Prerequisites
+
+1. **JDK** 11
+2. **IntelliJ** IDEA
+3. **Gradle** 6.2
+
+### 2.2. Setting up the project locally
+
+1. **Fork** this repo, and **clone** the fork into your computer.
+2. Open IntelliJ (if you are not in the welcome screen, click **`File`** > **`Close Project`** to close the existing project dialog first).
+3. Set up the correct JDK version for Gradle  
+   a.To set up the correct project structure **`Configure`** > **`Project Defaults`** > **`Project Structure`**  
+   b. Under **`New...`** find the directory of the appropriate JDK version.
+4. Click **`Import Project`**.
+5. Find the **`build.gradle`** file and select it. Click **`OK`**.
+6. Click **`Open as Project`**.
+7. Click **`OK`** and accept the default settings.
+
+### 2.3. Configure the coding style
+
+In IntelliJ's IDEA we adopt [[se-edu/guides] IDEA: Configuring the code style](https://se-education.org/guides/tutorials/intellijCodeStyle.html)
+to set up IDEA’s coding style to match ours.
+
+> Optionally, you can follow the guide [[se-edu/guides] Using Checkstyle](https://se-education.org/guides/tutorials/checkstyle.html)
+> to find how to use the CheckStyle within IDEA e.g., to report problems as you write code.
+
+## 3. Design
+
+<!-- {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.} -->
+
+### 3.1 Architecture
+
+**How the architecture components interact with each other**
+
+Core program flow is managed by the Duke class.
+
+![Main Program Flow](images/mainProgramFlow.png)
+
+The Duke class delegates work to the Ui class to handle user input.
+
+User input is passed to the Parser class to parse the input as a command.
+
+Each command subclass handles its own execution.
+
+<code>Storage</code>: Reads and writes data to the hard disk in a NUSMods export link format.
+
+### 3.2 Model Component
+
+The Model component is responsible for dealing with any module related data. It consists of the following classes:
+
+- `Day`: Represents a day in the timetable.
+- `LessonType`: Represents the type of lesson.
+- `Module`: Represents a module in NUS.
+- `ModuleLoader`: Loads the module information from the NUSMods resource file.
+- `RawLesson`: Represents a single block in a timetable representing one lesson slot. Properties are meant to be
+  freely accessed but not modified.
+- `SelectedModule`: Represents a module selected by the user that is to be added into his or her timetable.
+- `SemesterData`: Semester data contains all the module information pertaining to a single semester.
+- `Timetable`: Creates a timetable for the user with their selected modules and in their planning semester.
+
+### 3.3 Parser Component
+
+![Parser Class](images/parserClass.png)  
+
+
+The <code>Parser</code> component can:
+
+- return the correct command type based on user input.
+
+### 3.4 Command Component
+
+![Command Abstract Class](images/commandClass.png)
```

Comment:
Consider the symbols beside the variables and methods, maybe you should check the right format to show visibility?
![image](https://user-images.githubusercontent.com/88614316/198250238-fbbd2174-7d74-472b-84aa-3da7a0590c49.png)